interop_binaries/src/lib.rs

```diff
@@ -687,6 +684,6 @@ pub mod test_util {
     }
 
     pub fn generate_unique_name(prefix: &str) -> String {
-        format!("{}_{}", prefix, hex::encode(random::<[u8; 4]>()))
+        format!("{}-{}", prefix, hex::encode(random::<[u8; 4]>()))
```

Comment:
FWIW, this is necessary to fix issues in the `end_to_end` test. I was seeing error messages in container logs saying that requests couldn't be made because hostnames were invalid. I think this happened because `testcontainers` now turns on the `reqwest/hickory-dns` feature, and Hickory is more strict about underscores not being allowed in hostnames.